diff --git a/src/core/lib/security/credentials/jwt/jwt_verifier.c b/src/core/lib/security/credentials/jwt/jwt_verifier.c
index 0e2a264371..5c1e6a70f7 100644
--- a/src/core/lib/security/credentials/jwt/jwt_verifier.c
+++ b/src/core/lib/security/credentials/jwt/jwt_verifier.c
@@ -493,6 +493,7 @@ static EVP_PKEY *pkey_from_jwk(grpc_exec_ctx *exec_ctx, const grpc_json *json,
     gpr_log(GPR_ERROR, "Could not create rsa key.");
     goto end;
   }
+#if OPENSSL_VERSION_NUMBER < 0x10100005L
   for (key_prop = json->child; key_prop != NULL; key_prop = key_prop->next) {
     if (strcmp(key_prop->key, "n") == 0) {
       rsa->n =
@@ -508,6 +509,28 @@ static EVP_PKEY *pkey_from_jwk(grpc_exec_ctx *exec_ctx, const grpc_json *json,
     gpr_log(GPR_ERROR, "Missing RSA public key field.");
     goto end;
   }
+#else
+  {
+    BIGNUM *bn_n = NULL;
+    BIGNUM *bn_e = NULL;
+    for (key_prop = json->child; key_prop != NULL; key_prop = key_prop->next) {
+      if (strcmp(key_prop->key, "n") == 0) {
+       bn_n =
+          bignum_from_base64(exec_ctx, validate_string_field(key_prop, "n"));
+       if (bn_n == NULL) goto end;
+      } else if (strcmp(key_prop->key, "e") == 0) {
+       bn_e =
+          bignum_from_base64(exec_ctx, validate_string_field(key_prop, "e"));
+       if (bn_e == NULL) goto end;
+      }
+    }
+    RSA_set0_key(rsa, bn_n, bn_e, NULL);
+    if (bn_e == NULL || bn_n == NULL) {
+      gpr_log(GPR_ERROR, "Missing RSA public key field.");
+      goto end;
+    }
+  }
+#endif
   result = EVP_PKEY_new();
   EVP_PKEY_set1_RSA(result, rsa); /* uprefs rsa. */

