diff --git a/CMakeLists.txt b/CMakeLists.txt
index 680ee8b..45c36f0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -54,16 +54,16 @@ else()
   set(gRPC_INSTALL OFF CACHE BOOL "Generate installation target")
 endif()

-set(gRPC_ZLIB_PROVIDER "module" CACHE STRING "Provider of zlib library")
+set(gRPC_ZLIB_PROVIDER "package" CACHE STRING "Provider of zlib library")
 set_property(CACHE gRPC_ZLIB_PROVIDER PROPERTY STRINGS "module" "package")

 set(gRPC_CARES_PROVIDER "module" CACHE STRING "Provider of c-ares library")
 set_property(CACHE gRPC_CARES_PROVIDER PROPERTY STRINGS "module" "package")

-set(gRPC_SSL_PROVIDER "module" CACHE STRING "Provider of ssl library")
+set(gRPC_SSL_PROVIDER "package" CACHE STRING "Provider of ssl library")
 set_property(CACHE gRPC_SSL_PROVIDER PROPERTY STRINGS "module" "package")

-set(gRPC_PROTOBUF_PROVIDER "module" CACHE STRING "Provider of protobuf library")
+set(gRPC_PROTOBUF_PROVIDER "package" CACHE STRING "Provider of protobuf library")
 set_property(CACHE gRPC_PROTOBUF_PROVIDER PROPERTY STRINGS "module" "package")

 set(gRPC_GFLAGS_PROVIDER "module" CACHE STRING "Provider of gflags library")
@@ -225,6 +225,7 @@ elseif("${gRPC_SSL_PROVIDER}" STREQUAL "package")
 endif()

 if("${gRPC_GFLAGS_PROVIDER}" STREQUAL "module")
+  set(REGISTER_INSTALL_PREFIX OFF )
   if(NOT GFLAGS_ROOT_DIR)
     set(GFLAGS_ROOT_DIR ${CMAKE_CURRENT_SOURCE_DIR}/third_party/gflags)
   endif()
@@ -13836,3 +13837,26 @@ foreach(_config gRPCConfig gRPCConfigVersion)
     DESTINATION ${CMAKE_INSTALL_CMAKEDIR}
   )
 endforeach()
+
+# create a debian package
+set (CPACK_GENERATOR "DEB")
+execute_process (COMMAND uname -m OUTPUT_VARIABLE __machine_arch
+  OUTPUT_STRIP_TRAILING_WHITESPACE
+  RESULT_VARIABLE rc)
+if (${__machine_arch} STREQUAL "x86_64")
+  set (CPACK_DEBIAN_PACKAGE_ARCHITECTURE "amd64")
+  set (CPACK_RPM_PACKAGE_ARCHITECTURE "x86_64")
+else()
+  set (CPACK_DEBIAN_PACKAGE_ARCHITECTURE "i386")
+  set (CPACK_RPM_PACKAGE_ARCHITECTURE "i386")
+endif()
+set (CPACK_PACKAGE_NAME "${PROJECT_NAME}-dev")
+set (CPACK_PACKAGE_FILE_NAME ${CPACK_PACKAGE_NAME}-${PACKAGE_VERSION}.${CPACK_DEBIAN_PACKAGE_ARCHITECTURE})
+set (CPACK_PACKAGE_VERSION "${PACKAGE_VERSION}")
+set (CPACK_PACKAGE_CONTACT "p4c@barefootnetworks.com")
+set (CPACK_PACKAGE_VENDOR "Google")
+set (CPACK_PACKAGE_DESCRIPTION_SUMMARY "An RPC library and framework")
+set (CPACK_PACKAGE_DESCRIPTION "An RPC library and framework")
+set (CPACK_DEBIAN_PACKAGE_DEPENDS "libssl1.0.0, protobuf-dev (>=3.2.0), zlib1g-dev")
+set (CPACK_RPM_PACKAGE_REQUIRES "libssl1.0.0, protobuf-devel (>=3.2.0), zlib1g-devel")
+include (CPack)
