diff --git a/bf-asm/walle/CMakeLists.txt b/bf-asm/walle/CMakeLists.txt
index 25fedd4b..c9c13f21 100644
--- a/bf-asm/walle/CMakeLists.txt
+++ b/bf-asm/walle/CMakeLists.txt
@@ -4,7 +4,7 @@ cmake_minimum_required (VERSION 3.0.2 FATAL_ERROR)
 project(WALLE)
 
 find_package (PythonInterp REQUIRED)
-find_program (PYINSTALLER pyinstaller)
+find_program (PYINSTALLER pyinstaller2)
 if (NOT PYINSTALLER)
   MESSAGE(FATAL_ERROR "Pyinstaller is required")
 endif()
diff --git a/bf-asm/walle/walle.py b/bf-asm/walle/walle.py
index 019bc16f..ea8c686b 100755
--- a/bf-asm/walle/walle.py
+++ b/bf-asm/walle/walle.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 """
 Walle - JSON-to-binary cruncher tool
 See README.md for usage instructions
diff --git a/p4c/CMakeLists.txt b/p4c/CMakeLists.txt
index 382a82ba..873d640f 100644
--- a/p4c/CMakeLists.txt
+++ b/p4c/CMakeLists.txt
@@ -53,10 +53,7 @@ set (P4C_LIB_DEPS)
 find_package (PythonInterp REQUIRED)
 find_package (FLEX REQUIRED)
 find_package (BISON REQUIRED)
-set(SAVED_CMAKE_FIND_LIBRARY_SUFFIXES ${CMAKE_FIND_LIBRARY_SUFFIXES})
-set(CMAKE_FIND_LIBRARY_SUFFIXES .a)
 find_package (Protobuf 3.0.0 REQUIRED)
-set(CMAKE_FIND_LIBRARY_SUFFIXES ${SAVED_CMAKE_FIND_LIBRARY_SUFFIXES})
 # The boost graph headers are optional and only required by the graphs backend
 find_package (Boost QUIET COMPONENTS graph)
 if (Boost_FOUND)
diff --git a/p4c/tools/driver/p4c.in b/p4c/tools/driver/p4c.in
index 09c3b0ea..71b30f6f 100644
--- a/p4c/tools/driver/p4c.in
+++ b/p4c/tools/driver/p4c.in
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 #
 # Copyright 2013-present Barefoot Networks, Inc.
 #
